jupyterhub/tests/test_singleuser.py

```diff
@@ -198,6 +197,70 @@ def assert_not_in_home(path, name):
                 assert_not_in_home(path, key)
 
 
+@pytest.mark.parametrize("extension", [True, False])
+@pytest.mark.parametrize("notebook_dir", ["", "~", "~/sub", "ABS"])
+async def test_notebook_dir(
+    request, app, tmpdir, user, full_spawn, extension, notebook_dir
+):
+    token = user.new_api_token(scopes=["access:servers!user"])
+    headers = {"Authorization": f"Bearer {token}"}
+
+    spawner = user.spawner
+    if extension:
+        user.spawner.environment["JUPYTERHUB_SINGELUSER_EXTENSION"] = "1"
```

Comment:
```suggestion
        user.spawner.environment["JUPYTERHUB_SINGLEUSER_EXTENSION"] = "1"
```
Alternatively since the `extension` param is only used in this conditional you could set the param to `["0", "1"]` instead of `[True, False]` and get rid of the `if`